ra/ra_test.go

```diff
@@ -1135,6 +1136,24 @@ func TestAuthzRateLimiting(t *testing.T) {
 	test.AssertNotError(t, err, "NewAuthorization failed")
 }
 
+func TestAuthzFailedRateLimiting(t *testing.T) {
+	_, sa, ra, _, cleanUp := initAuthorities(t)
+	defer cleanUp()
+
+	ra.rlPolicies = &dummyRateLimitConfig{
+		InvalidAuthorizationsPerAccountPolicy: ratelimit.RateLimitPolicy{
+			Threshold: 1,
+			Window:    cmd.ConfigDuration{Duration: 1 * time.Hour},
+		},
+	}
+
+	// override with our mockInvalidAuthorizationsAuthority to for specific test
+	ra.SA = sagrpc.NewStorageAuthorityClient(&mockInvalidAuthorizationsAuthority{})
+	// Should trigger rate limit
+	_, err := ra.NewAuthorization(ctx, AuthzRequest, Registration.ID)
+	test.AssertError(t, err, "Too many failed authorizations recently.")
```

Comment:
Based on the third argument to `AssertError` I think you might have misunderstood this test function slightly (which makes sense, it's a little awkward).

The third argument isn't the expected error message, it's the message printed if the second argument is `nil`. I recommend that you change the third argument to something like `"NewAuthorization did not encounter expected rate limit error"` and then add a subsequent `test.AssertEquals(t, err.Error(), "Too many failed authorizations recently.")` to check that the error had the expected message.